service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadDocumentsController.java

```diff
@@ -12,19 +12,31 @@
 import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
 import uk.gov.hmcts.reform.fpl.model.CaseData;
 import uk.gov.hmcts.reform.fpl.service.DocumentsValidatorService;
+import uk.gov.hmcts.reform.fpl.service.UploadDocumentsService;
+
+import java.util.List;
 
 @Api
 @RestController
 @RequestMapping("/callback/upload-documents")
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class UploadDocumentsController extends CallbackController {
     private final DocumentsValidatorService documentsValidatorService;
+    private final UploadDocumentsService uploadDocumentsService;
 
     @PostMapping("/mid-event")
     public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackrequest) {
+        CaseData caseDataBefore = getCaseDataBefore(callbackrequest);
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
         CaseData caseData = getCaseData(caseDetails);
 
-        return respond(caseDetails, documentsValidatorService.validateDocuments(caseData));
+        List<String> errors = documentsValidatorService.validateDocuments(caseData);
+
+        if (errors.isEmpty()) {
+            caseDetails.getData().putAll(uploadDocumentsService
+                .updateCaseDocuments(caseDataBefore, caseData));
```

Comment:
again please consider reversing params.

Reply:
I have done reversing params this time but as I was saying to you, I don't agree with your reasoning and I believe this should be left to the opinion of whoever is working on it.